aiohttp_session/redis_storage.py

```diff
@@ -19,17 +19,17 @@ def __init__(self, redis_pool, *, cookie_name="AIOHTTP_SESSION",
         self._encoder = encoder
         self._decoder = decoder
         self._key_factory = key_factory
+        assert isinstance(redis_pool, aioredis.commands.Redis), redis_pool
```

Comment:
Let's left `assert` and `AssertionError` to unittests, explicit `TypeError` is much better.

Reply:
Would you agree if I replaced this with:
```py
if isinstance(redis_pool, aioredis.pool.ConnectionsPool):
    warnings.warn(
        "using a pool created with aioredis.create_pool is deprecated"
        "please provide a pool created with aioredis.create_redis_pool",
        DeprecationWarning
    )
    redis_pool = aioredis.commands.Redis(redis_pool)
elif not isinstance(redis_pool, aioredis.commands.Redis):
    raise TypeError("Expexted aioredis.commands.Redis got {}".format(type(redis_pool)))
```

This way we keep compatibility with the old `create_pool` function, allowing for a smoother transition of the users. (It should work from a look I had in `aioredis` source but to be honest I haven't tested it yet)

Reply:
I'll include the above changes in my next commit. If you disagree with that implementation / approach feel free to drop me a line and I'll revert them :)